extensions/emoji/src/emoji.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  ActionPanel,
+  CopyToClipboardAction,
+  PasteAction,
+  List,
+  getLocalStorageItem,
+  setLocalStorageItem,
+} from "@raycast/api";
+
+import { useState, useEffect, useRef, useCallback } from "react";
+import type { ReactElement, SetStateAction, Dispatch } from "react";
+import { createEmojiList } from "generate-emoji-list";
+
+type Category = { category: string; emojis: Emoji[] };
+type Emoji = {
+  emoji: string;
+  description: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const useStateFromLocalStorage = <T, _ = void>(key: string, initialValue: T): [T, Dispatch<SetStateAction<T>>] => {
+  const [state, setState] = useState<T>(initialValue);
+  const didUnmount = useRef<boolean>(false);
+
+  useEffect(() => {
+    (async () => {
+      const cache = await getLocalStorageItem(key);
+
+      if (typeof cache === "string") {
+        if (!didUnmount.current) {
+          setState(JSON.parse(cache));
+        }
+      }
+    })();
+
+    return () => {
+      didUnmount.current = true;
+    };
+  }, []);
+
+  const setStateAndLocalStorage = useCallback((updater) => {
+    setState((state) => {
+      const newValue = typeof updater === "function" ? updater(state) : updater;
+      setLocalStorageItem(key, JSON.stringify(newValue));
+      return newValue;
+    });
+  }, []);
+
+  return [state, setStateAndLocalStorage];
+};
+
+export default function Main(): ReactElement {
+  const [searchText, setSearchText] = useState<string>("");
+  const [list, setList] = useStateFromLocalStorage<Category[]>("emoji-list", []);
+  const didUnmount = useRef<boolean>(false);
+
+  useEffect(() => {
+    createEmojiList().then((list: Category[]) => {
+      if (!didUnmount.current) {
+        setList(list);
+      }
+    });
+
+    return () => {
+      didUnmount.current = true;
+    };
+  }, []);
+
+  const [recentlyUsed, setRecentlyUsed] = useStateFromLocalStorage<Emoji[]>("recently-used", []);
+  const addToRecentlyUsed = useCallback((emoji: Emoji) => {
+    setRecentlyUsed((list) =>
+      list.find((x) => x.description === emoji.description) ? list : [emoji, ...list].slice(0, 10)
+    );
+  }, []);
+
+  return (
+    <List onSearchTextChange={setSearchText} isLoading={list.length === 0}>
+      {[{ category: "Recently Used", emojis: recentlyUsed }, ...list].map((category: Category) => (
+        <List.Section title={category.category} key={category.category}>
+          {category.emojis
+            .filter((emoji) => emoji.description.includes(searchText))
```

Comment:
As far as I can see you are using the custom search which is completely similar to what we are doing in the built-in one. 
I'd recommend removing `onSearchTextChange` a filtering logic completely. It makes the searching works not smoothly.

And by the way, in the case, you need to provide special characters for the search, check out the `keywords` property for [List.Item](https://developers.raycast.com/api-reference/user-interface/list#props-1)